internal/k8s/reconciler/listener.go

```diff
@@ -47,12 +48,23 @@ type K8sListenerConfig struct {
 	ConsulNamespace string
 	Logger          hclog.Logger
 	Client          gatewayclient.Client
+	State           *state.ListenerState
```

Comment:
I'm assuming this is unused? Looks like `ListenerState` is what you're using elsewhere.

Reply:
This is `K8sListenerConfig.State` (arg to the constructor) vs. `K8sListener.ListenerState`

Reply:
So, it looks like it's not set at all in the constructor diff starting at line 57, so still thinking it's unused. Either way, if it isn't unused, these should be collapsed into a single field, since there should only ever be a single state associated with a listener:

```go
	lState := config.State
	if lState == nil {
		lState = &state.ListenerState{
			Name:     listener.Name,
			Protocol: listener.Protocol,
			Routes:   make(map[string]core.ResolvedRoute),
		}
	}

	return &K8sListener{
		ListenerState:   lState,
```

Reply:
oh gosh, nvm, the diff above was confusing me, disregard this, I see what you're saying